format-avro/src/test/java/io/cdap/plugin/format/avro/input/AvroInputFormatProviderTest.java

```diff
@@ -70,4 +71,41 @@ public void testSchemaDetection() throws IOException {
     Schema schema = schemaDetector.detectSchema(avroFile.getPath(), formatContext, Collections.emptyMap());
     Assert.assertEquals(expectedSchema, schema);
   }
+
+  @Test
+  public void testPathFieldSchemaDetection() throws IOException {
+    // write avro file
+    boolean existingPathFieldValue = false;
+    try {
+      Schema expectedSchema = Schema.recordOf("Avro",
+        Schema.Field.of("id", Schema.of(Schema.Type.INT)),
+        Schema.Field.of("name", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+        Schema.Field.of("score", Schema.of(Schema.Type.DOUBLE)),
+        Schema.Field.of("test", Schema.of(Schema.Type.STRING)));
+      org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(expectedSchema.toString());
+      GenericRecord record = new GenericRecordBuilder(avroSchema)
+        .set("id", 0)
+        .set("name", "alice")
+        .set("score", 1.0d)
+        .set("test", "value")
+        .build();
+
+      File avroFile = new File(TMP_FOLDER.newFolder(), "test.avro");
+      DatumWriter<GenericRecord> datumWriter = new GenericDatumWriter<>(avroSchema);
+      DataFileWriter<GenericRecord> dataFileWriter = new DataFileWriter<>(datumWriter);
+      dataFileWriter.create(avroSchema, avroFile);
+      dataFileWriter.append(record);
+      dataFileWriter.close();
+
+      // call detect schema
+      AvroInputFormatProvider.Conf conf = new AvroInputFormatProvider.Conf("test");
+      AvroInputFormatProvider formatProvider = new AvroInputFormatProvider(conf);
+      FormatContext formatContext = new FormatContext(new MockFailureCollector(), null);
+      SchemaDetector schemaDetector = new SchemaDetector(formatProvider);
+      Schema schema = schemaDetector.detectSchema(avroFile.getPath(), formatContext, Collections.emptyMap());
+    } catch (Exception e) {
```

Comment:
So this is expected to throw an exception? I think it would be better to explicitly verify the exception that is being thrown.

Reply:
After looking at the rest of the PR, detectSchema() should not be throwing an exception. Instead, you should be checking the failure from the FailureCollector and validating that it is correct here.

Reply:
Changed to checking if failure collector  is empty